src/main/java/randoop/reflection/OperationExtractor.java

```diff
@@ -38,55 +39,117 @@
   private final Collection<TypedOperation> operations;
 
-  /** The class type of the declaring class for the collected operations */
+  /** The type of the declaring class for the collected operations */
   private ClassOrInterfaceType classType;
 
+  /** The predicate to test whether to omit an operation */
+  private OmitMethodsPredicate omitPredicate;
+
   /**
    * Creates a visitor object that collects the {@link TypedOperation} objects corresponding to
-   * members of the class type and satisfying the given predicate.
+   * members of the class satisfying the given visibility and reflection predicates and that don't
+   * violate the omit method predicate.
+   *
+   * <p>Once created this visitor should only be applied to members of {@code
+   * classType.getRuntimeType()}.
    *
-   * @param classType the declaring classtype for collected operations
-   * @param operations the collection of operations
+   * @param classType the declaring class for collected operations
+   * @param operations the collection of operations, will be side-effected
    * @param predicate the reflection predicate
+   * @param omitPredicate the list of {@code Pattern} objects for omitting methods, may be null
    * @param visibilityPredicate the predicate for test visibility
    */
   public OperationExtractor(
       ClassOrInterfaceType classType,
       Collection<TypedOperation> operations,
       ReflectionPredicate predicate,
+      OmitMethodsPredicate omitPredicate,
       VisibilityPredicate visibilityPredicate) {
     this.classType = classType;
     this.operations = operations;
     this.predicate = predicate;
     this.visibilityPredicate = visibilityPredicate;
+    this.omitPredicate = omitPredicate;
+  }
+
+  /**
+   * Creates a visitor object that collects the {@link TypedOperation} objects corresponding to
+   * members of the class satisfying the given visibility and reflection predicates.
+   *
+   * @param classType the declaring class for collected operations
+   * @param operations the collection of operations, will be side-effected
+   * @param predicate the reflection predicate
+   * @param visibilityPredicate the predicate for test visibility
+   */
+  public OperationExtractor(
+      ClassOrInterfaceType classType,
+      Collection<TypedOperation> operations,
+      ReflectionPredicate predicate,
+      VisibilityPredicate visibilityPredicate) {
+    this(classType, operations, predicate, OmitMethodsPredicate.NO_OMISSION, visibilityPredicate);
   }
 
   /**
-   * Adds an operation to the collection of this extractor. If the declaring class type is an {@link
-   * InstantiatedType}, then the substitution for that class is applied to the types of the
-   * operation, and this instantiated operation is returned.
+   * Adds an operation to the collection of this extractor.
    *
    * @param operation the {@link TypedClassOperation}
    */
   private void addOperation(TypedClassOperation operation) {
     if (operation != null) {
```

Comment:
At every call site, the argument is non-null.  So, make this an assertion, or remove it.  (Alternately, add documentation that the formal parameter can be null and what happens in that case, but I don't see the point of doing so.)